moto/sns/models.py

```diff
@@ -302,6 +302,7 @@ def _field_match(field, rules, message_attributes):
                         return False
 
                     for attribute_values in attribute_values:
+                        attribute_values = float(attribute_values)
```

Comment:
This for-loop should probably say `for attribute_value in attribute_values`. It's currently re-using the same variable-name, that's never a good idea.

(I know it's not your change, but I suddenly noticed it, and we may as well fix it now.)